src/scimPatch.ts

```diff
@@ -401,6 +402,21 @@ function assign(obj:any, keyPath:Array<string>, value:any) {
         }
         obj = obj[key];
     }
+
+    // If the attribute is an array and the operation is "add",
+    // then the value should be added to the array
+    const attribute = obj[keyPath[lastKeyIndex]];
+    if (op.toLowerCase() === "add" && Array.isArray(attribute)) {
+        // If the value is also an array, append all values of the array to the attribute
+        if (Array.isArray(value)) {
+            value.forEach((val) => attribute.push(val));
+            return;
+        }
+
+        // If value is not an array, simply append it as a whole to end of attribute
+        attribute.push(value);
```

Comment:
Same as the other, I'll prefer to affect directly into the array.
```suggestion
        obj[keyPath[lastKeyIndex]] = [...attribute, value];
```